File: Documentation/API/Utilities/CalculateTiming.js
import RhythmGameUtilities from '@rhythm-game-utilities/core';

const seconds = 2;
const resolution = 192;
const positionDelta = 50;

const tempoChanges = [{ position: 0, bpm: 120000 }];

const currentPosition = RhythmGameUtilities.ConvertSecondsToTicks(seconds, resolution, tempoChanges);

const timing = RhythmGameUtilities.CalculateTiming(750, currentPosition, positionDelta);

console.log(timing); // Hit

File: Documentation/API/Utilities/CalculateTiming.lua
---@type RhythmGameUtilities
local rhythmgameutilities = require("rhythmgameutilities")

local seconds = 2;
local resolution = 192;
local position_delta = 50;

local tempo_changes = { { position = 0, bpm = 120000 } };

local current_position =
    rhythmgameutilities.convert_seconds_to_ticks(seconds, resolution, tempo_changes);

local value = rhythmgameutilities.calculate_timing(750, current_position, position_delta);

print(value) -- Hit
